mailing_list/models.py

```diff
@@ -122,16 +122,21 @@ def __unicode__(self):
     def _get_unsubscribed_email_set(self):
         return {x.email for x in self.unsubscribed_set.all()}
 
-    def _get_enrolled_email_set(self):
+    def _get_enrolled_persons(self):
         univ_ids = []
-        canvas_enrollments = canvas_api_client.get_enrollments(self.canvas_course_id, self.section_id)
+        canvas_enrollments = canvas_api_client.get_enrollments(
+                                 self.canvas_course_id, self.section_id)
         for enrollment in canvas_enrollments:
             try:
                 univ_ids.append(enrollment['user']['sis_user_id'])
             except KeyError:
-                logger.debug("Found canvas enrollment with missing sis_user_id %s", json.dumps(enrollment, indent=4))
+                logger.debug(
+                    "Found canvas enrollment with missing sis_user_id %s",
+                    json.dumps(enrollment, indent=4))
+        return list(Person.objects.filter(univ_id__in=univ_ids))
 
-        return {p.email_address for p in Person.objects.filter(univ_id__in=univ_ids)}
```

Comment:
Can we keep the original here, returning the generator instead of the fleshed out list?